### PR TITLE
Build Portal Updates

### PR DESCRIPTION
**Pull request by StephenHodgson**
Overview
---
- Added quick options to switching between Immersive Head Mounted Displays (IMHDs) and HoloLens
-  Manually add IP of remote devices
-  Install via WiFi
-  Better enumeration of Appx Package folders. #1136
-  Fixed Build window incrementing the reserved number for Windows Store Apps #1402
-  Add support for installing on local machine (for IMHD apps)
-  HTTPS support #98

image
image
image

Changes
---
- Fixes: #1020
- Fixes: #98 
- Fixes: #1402 
- Fixes: #1424
- Fixes: #1519

### CODE REVIEW

**Comment by StephenHodgson** [comment]
@NeerajW sorry I had to merge some conflicts.

File: Assets/HoloToolkit/BuildAndDeploy/Editor/BuildDeployTools.cs [hunk 1]
```diff
@@ -230,14 +241,17 @@ public static bool BuildAppxFromSLN(string productName, string msBuildVersion, b
             }
 
             string nugetPath = Path.Combine(unity, @"Data\PlaybackEngines\MetroSupport\Tools\NuGet.exe");
+            string assemblyCSharp = storePath + "/GeneratedProjects/UWP/Assembly-CSharp";
+            string assemblyCSharpFisrtpass = storePath + "/GeneratedProjects/UWP/Assembly-CSharp-firstpass";
```
Comment (keveleigh) [inline Assets/HoloToolkit/BuildAndDeploy/Editor/BuildDeployTools.cs]: typo in the variable name: `Fisrtpass`

Comment (StephenHodgson) [reply Assets/HoloToolkit/BuildAndDeploy/Editor/BuildDeployTools.cs]: Exactly why I hate strings 😆

**Comment by DDReaper** [comment]
Found 1 issue.
If I add a new machine on the Deploy options and then select Local Machine. The "Open Device Portal" button is still enabled.  Which doesn't work as it points to 0.0.0.0?

Although in fairness, this should be available and point to https://localhost:50443

Thoughts @StephenHodgson ?

**Comment by StephenHodgson** [comment]
Well, if you select local machine, then the active IP is the local.  But I should look at cleaning that up if the address isn't properly filled out. 

Under the hood if the address is invalid, it doesn't do anything anyway.

**Comment by DDReaper** [comment]
50443 is the default port assigned for Immersive Headsets I guess. Unless the port is totally random, in which case you'll need to make it configurable on the screen.  Best to get a few to test and report their ports

**Comment by StephenHodgson** [comment]
Why would you need to add an immersive headset to the machine options?
The local machine is the one for a local build with the immersive headsets, and it installs and runs the appx via powershell.

**Comment by DDReaper** [comment]
Because the Device Portal is a general device portal and also works for immersive headsets :D

**Comment by StephenHodgson** [comment]
I'm confused.  Why do you need the device portal for a local PC machine though?
You just run, install, and do what you need right in the environment you're in.

**Comment by StephenHodgson** [comment]
The Build window already handles everything you need to do for the local machine

**Comment by DDReaper** [comment]
Ok, fair enough.  Only thought Immersive / Local machine users might find easy access to the device portal useful as well.  If we're not supporting that, then just ensure the button is only enabled for IP / HL targets :D

**Comment by StephenHodgson** [comment]
> If we're not supporting that, then just ensure the button is only enabled for IP / HL targets :D

I believe it should get disabled if there isn't a HoloLens detected.  Let me know if this isn't the case.

**Comment by NeerajW** [comment]
@DDReaper added you as moderator. Please accept invite.
Appreciate your review here.

**Pull request merged**